drivers/bus/isapnp/pdo.c

```diff
@@ -137,6 +137,57 @@ IsaPdoQueryId(
     return STATUS_SUCCESS;
 }
 
+NTSTATUS
+IsaPdoQueryDeviceText(
+    _In_ PISAPNP_PDO_EXTENSION PdoExt,
+    _Inout_ PIRP Irp,
+    _In_ PIO_STACK_LOCATION IrpSp)
+{
+    NTSTATUS Status;
+    PWCHAR Buffer;
+    size_t CharCount;
+
+    switch (IrpSp->Parameters.QueryDeviceText.DeviceTextType)
+    {
+        case DeviceTextDescription:
+        {
+            CharCount = strlen(PdoExt->IsaPnpDevice->FriendlyName) +
+                        sizeof(ANSI_NULL);
+
+            if (CharCount == 1)
+                return STATUS_NOT_SUPPORTED;
+
+            Buffer = ExAllocatePoolWithTag(PagedPool,
+                                           CharCount * sizeof(WCHAR),
+                                           TAG_ISAPNP);
+            if (!Buffer)
+                return STATUS_INSUFFICIENT_RESOURCES;
+
+            Status = RtlStringCchPrintfExW(Buffer,
```

Comment:
Why don't you just use RtlStringCchPrintfW() then? (since all the other parameters are NULL or 0)
Or RtlAnsiStringToUnicodeString() ?

Reply:
> Why don't you just use RtlStringCchPrintfW() then?

I want to reduce the driver's physical memory consumption because all the strsafe functions are implemented in header file.